tools/mlir-gen/MLIRGen.cpp

```diff
@@ -375,16 +378,34 @@ Value MLIRGenerator::lowerNamedMatmul(Value input, Value weight, Value output) {
 
 Value MLIRGenerator::lowerMatmul(Value input, Value weight, Value output) {
   Value chain;
-  auto inputShape = cast<ShapedType>(input.getType());
-  auto outputShape = cast<ShapedType>(output.getType());
+  auto inputType = cast<ShapedType>(input.getType());
+  auto outputType = cast<ShapedType>(output.getType());
+
+  if (vnniPacked) {
+    SmallVector<int64_t> vnniShape{inputType.getShape()};
+    vnniShape.back() = vnniShape.back() / vnniFactor;
+    vnniShape.push_back(vnniFactor);
+    auto vnniType =
```

Comment:
shouldn't we also make sure that this is the same as the `weight` shape?

Reply:
Assert wouldn't hurt